core/src/main/scala/chisel3/VerificationStatement.scala

```diff
@@ -41,7 +41,35 @@ object assert {
   )(
     implicit sourceInfo: SourceInfo,
     compileOptions:      CompileOptions
-  ): Assert = macro _applyMacroWithMessage
+  ): Assert = macro _applyMacroWithStringMessage
+
+  /** Checks for a condition to be valid in the circuit at all times. If the
```

Comment:
First of all thanks for adding this scaladoc!

Some small suggestions for improvement:

> Checks for a condition to be valid in the circuit at all times.

Technically the condition is only checked at a rising clock edge. If there was an asynchronous signal being fed into the assert, then a change in that signal might be missed.

> Does not fire when in reset (defined as the encapsulating Module's reset).
I would suggest to remove the explanation in parenthesis since the reset could also be changed through `withReset`.

One other thing that might be worth adding would be a short explanation of the behavior when used not with simulation, but with a bounded model checker like Jasper Gold or the formal backend in chiseltest.

Reply:
Would it be better to change the reset condition doc to also add `withReset`? Something like

> Does not fire when in reset, where reset is defined as the encapsulating Module's reset or that of the encapsulating `withReset` block

Reply:
> Would it be better to change the reset condition doc to also add `withReset`? Something like

Personally I feel like having a single place where we explain how resets work in Chisel would be preferable. Then for specific functions we can just write something like `Does not fire when reset is active.` with maybe a link to the central explanation of how implicit `reset` works in Chisel.

Reply:
Agreed